dockerfile

```diff
@@ -0,0 +1,74 @@
+FROM ubuntu:24.04 as apt-requirements
+
+# Install APT requirements.
+RUN --mount=target=/requirements,type=bind,source=requirements <<EOF
+
+    set -ex
+
+    apt update
+
+    apt --yes --no-install-recommends install $(grep -vE "^\s*#" /requirements/requirements.system.txt  | tr "\n" " ")
+
+    apt clean && rm -rf /var/lib/apt/lists/*
+
+EOF
+
+# Compile Google Test.
+FROM apt-requirements as compile-google-test
+
+ARG GOOGLETEST_VERSION
+
+RUN <<EOF
+
+    set -ex
+
+    git clone --depth=1 --branch=v${GOOGLETEST_VERSION} https://github.com/google/googletest googletest
+
+    cd googletest
+
+    cmake -S . -B build -DCMAKE_INSTALL_PREFIX=/opt/google-test-${GOOGLETEST_VERSION}
+
+    cmake --build build
+
+    cmake --build build --target install
+
+EOF
+
+# Compile Kokkos.
+FROM apt-requirements as compile-kokkos
+
+ARG KOKKOS_VERSION
+ARG KOKKOS_PRESET
+
+COPY kokkos.cmake.presets.json /kokkos.cmake.presets.json
```

Comment:
We could also create a `presets` directory, put the `kokkos.cmake.presets.json` there, and then use the `mount` like you did for google test.

Reply:
Well, in this case, the `COPY` does not hurt, because in the final image it will not be copied (only the `/opt/kokkos-<version>/<preset>` will be)...